src/main/java/Bag.java

```diff
@@ -0,0 +1,60 @@
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public interface Bag<T> {
+
+    /**
+     * Add an element to the bag.
+     *
+     * @param element item to be added to the bag
+     */
+    void add(T element);
```

Comment:
It's customary for `add` to return a `boolean` indicating if the item was actually added or not. We'd likely always return `true` here, so fine to leave it as `void`.